src/helpers/paths.ts

```diff
@@ -23,23 +23,43 @@ export const getConfigFromFile = (configPath?: string): object => {
   return JSON.parse(rawConfig);
 };
 
-export const getModelsPaths = (basePath: string, extension: "js" | "ts"): string[] => {
-  const { base: basePathEnd } = path.parse(basePath);
-  const searchPath =
-    basePathEnd === "models" ? `**/!(index).${extension}` : `**/models/!(index).${extension}`;
-  const modelsFolderPath = path.join(basePath, searchPath);
+export const getModelsPaths = (basePath: string | undefined, extension: "js" | "ts"): string[] => {
+  if (basePath && basePath !== "") {
+    // base path, only check that path
+    const { ext } = path.parse(basePath);
+
+    // if path points to a folder, search all files not named index.ts or index.js in that folder.
```

Comment:
In my experience `index.{ts|js}` tend to be the one file in models folders that don't define their own Mongoose models. Would you say otherwise?

Reply:
Yes, I don't really see a reason why this would be the case _in general_.
This behavior should definitely be documented (although that might already be the case), but preferably I'd remove it altogether and let the user provide their own glob pattern. This prevents any unexpected behavior and increases flexibility for users.

(Of course it'd be fine to have "no index.*" as the default, but it should be documented)

Reply:
Yeah allowing a user to provide their own glob pattern wouldn't be a bad idea, it would enable support for mongoose models in different folders of a project (although seems like kind of an anti-pattern to me).

The index.* ignore could be removed but it should then silently skip files if they don't contain mongoose schemas - the current behaviour would throw an error because it expects all files read to export mongoose schemas.

Mongoose folder structures mostly all fit this file structure logic, if it became clear that there is a gap in logic here then adding the additional flexiblity would be more pressing but for now this seems fine to me. I'll add some docs about it for now.

Reply:
> The index.* ignore could be removed but it should then silently skip files if they don't contain mongoose schemas - the current behaviour would throw an error because it expects all files read to export mongoose schemas.

This kind of behavior would have to be implemented for #12 too right?

> for now this seems fine to me. I'll add some docs about it for now

Yeah that's probably the best thing to do for now.

Reply:
> This kind of behavior would have to be implemented for #12 too right?

Correct!